### PR TITLE
collaborator change handler on backend enforces roles …

### PR DESCRIPTION
**Pull request by njackson**
also, the socket acquires the user record from the sender's cookie

### CODE REVIEW

**Comment by njackson** [comment]
permissions

File: lib/services/sockets/manager.coffee [hunk 1]
```diff
@@ -83,8 +99,14 @@ class Manager
     redis = require 'socket.io-redis'
     io = sockets server
     io.adapter redis()
-
     @io = io.of @boardsNamespace()
+
+    @io.use (socket, next) ->
+      session = cookies.readSession('boardroom', 'a7c6dddb4fa9cf927fc3d9a2c052d889', 1000 * 60 * 60 * 24, socket.request)
```
Comment (sidke) [inline lib/services/sockets/manager.coffee]: moving this out to config somewhere might be a better idea than having it hardcoded twice but also having it hardcoded at all

File: assets/js/handler.coffee [hunk 2]
```diff
@@ -265,6 +266,10 @@ class boardroom.Handler
       return
     card.group().cards().remove card, { rebroadcast: true }
 
+  onDisallowedUpdate: (message) =>
+    alert("Action Forbidden!")
```
Comment (sidke) [inline assets/js/handler.coffee]: i will take care of moving this over to a modal

File: lib/services/sockets/manager.coffee (hunk 1 above)
Comment (sidke) [inline lib/services/sockets/manager.coffee]: it's also hardcoded here: https://github.com/carbonfive/boardroom/blob/development/lib/config.coffee#L18

**Comment by sidke** [comment]
LGTM

this gif has nothing to do with thumbs ups, i just ran out of kpop thumbs up gifs on giphy

**Pull request merged**